### PR TITLE
[stag] GOV-1233 Fix Tag based evaluation for asset

### PR DESCRIPTION
**Pull request by nikhilbonte21**

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ void authorizeResourceUpdate(AtlasEntity resourceEntity, AtlasVertex ResourceVer
                 if (atlasVertexIterator.hasNext()) {
                     //Found linked asset in store
                     AtlasVertex assetVertex = (AtlasVertex) atlasVertexIterator.next();
-                    assetEntity = entityRetriever.toAtlasEntityHeader(assetVertex);
+                    assetEntity = new AtlasEntityHeader(entityRetriever.toAtlasEntity(assetVertex));
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java:79]: Use function `toAtlasEntityHeaderWithClassifications` to fetch entities, why I'm saying, fetching whole entitiy sometimes slows it down if large no of relation present, which might be escalated later on. We should avoid using `toAtlasEntity` function and use alternative whenever possible.

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java [hunk 2]
```diff
@@ -101,7 +101,7 @@ void authorizeResourceDelete(AtlasVertex resourceVertex) throws AtlasBaseExcepti
 
             AtlasObjectId asset = getAssetRelationAttr(resourceEntity);
             if (asset != null) {
-                AtlasEntityHeader assetEntity = entityRetriever.toAtlasEntityHeader(asset.getGuid());
+                AtlasEntityHeader assetEntity =  new AtlasEntityHeader(entityRetriever.toAtlasEntity(asset.getGuid()));
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java:104]: USe `toAtlasEntityHeaderWithClassifications`

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ void authorizeResourceUpdate(AtlasEntity resourceEntity, AtlasVertex ResourceVer
             if (asset != null) {
                 //Found linked asset in payload
                 AtlasVertex assetVertex = entityRetriever.getEntityVertex(asset);
-                assetEntity = entityRetriever.toAtlasEntityHeader(assetVertex);
+                assetEntity = new AtlasEntityHeader(entityRetriever.toAtlasEntity(assetVertex));
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java:65]: Use `toAtlasEntityHeaderWithClassifications`

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java (hunk 1 above)
Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java]: Nice catch, I agree

**Review by mehtaanshul — APPROVED** [review]

**Pull request merged**